packages/office-ui-fabric-react/src/components/TextField/TextField.types.ts

```diff
@@ -163,17 +163,20 @@ export interface ITextFieldProps extends React.AllHTMLAttributes<HTMLInputElemen
 
   /**
    * Callback for when the input value changes.
+   * This is called on both `input` and `change` native events.
    */
   onChange?: (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => void;
 
   /**
+   * Use `onChange` instead.
    * @deprecated Use `onChange` instead.
    */
   onChanged?: (newValue: any) => void;
 
   /**
    * Called after the input's value updates but before re-rendering.
    * @param newValue - The new value. Type should be string.
+   * @deprecated Use `onChange`. Please file an issue if you have a use case `onChange` does not cover.
    */
   onBeforeChange?: (newValue: any) => void;
```

Comment:
There is a `_warnDeprecations` utility that can be called for both `onBeforeChange` and `onChanged`. Also, similar to the other warnings, there should be a communication that these props have been deprecated and will cause existing unit tests to fail if used. 

Side note: Adding these calls will cause any of our own unit tests that use those props to fail. Those tests should be moved to the `deprecated.test` file which automatically handles mocking out `warn` so that it doesn't causes tests to fail.

Reply:
Example: https://github.com/OfficeDev/office-ui-fabric-react/blob/fbef905cf9479a0cacb9fb0046826c1403286852/packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.base.tsx#L122

Good suggestion @JasonGore